### DISCUSSION TITLE
Random expander

### DISCUSSION THREAD

**Discussion by fcrozatier** [networkx/networkx#6494] [Algorithms]
There is no utility function to generate a random [expander](https://en.wikipedia.org/wiki/Expander_graph) graph. I could fill a PR if there is interest.

Suggestion: It has been [proved](https://arxiv.org/pdf/cs/0405020.pdf) that a random k-regular graph made of k/2 distinct permutations on {1,...,n} has probabilty $1 − O(n^{-τ})$ of being an expander, where $\tau$ is computable. 
We could add a utility `maybe_expander(n,k)` that returns such a graph and the probability of it being an expander. For engineers this utility would be enough. 

For mathematicians we could add an `is_regular_expander` utility and a slower `random_expander(n,k)` that loops over `maybe_expander` while not `is_regular_expander`. The `is_regular_expander` utility would be based on [Cheeger inequalities](https://en.wikipedia.org/wiki/Expander_graph#Cheeger_inequalities) for k-regular graphs (and the `is_regular` utility)

**Comment by rossbar** [comment]
Closing in favor of the issue.
